CMakeLists.txt

```diff
@@ -539,10 +550,12 @@ foreach(nf ${NO_USE_FILES})
 endforeach()
 
 # test section
-if(OCOS_ENABLE_CTEST AND OCOS_ENABLE_SELECTED_OPLIST)
-  # currently the tests don't handle operator exclusion cleanly.
-  message(WARNING "Due to usage of OCOS_ENABLE_SELECTED_OPLIST excluding operators the tests are unable to be built and run")
-elseif(OCOS_ENABLE_CTEST AND NOT OCOS_ENABLE_SELECTED_OPLIST)
+if(OCOS_ENABLE_CTEST)
+  if (OCOS_ENABLE_SELECTED_OPLIST)
+    # currently the tests don't handle operator exclusion cleanly.
+    message(FATAL "Due to usage of OCOS_ENABLE_SELECTED_OPLIST excluding operators the tests are unable to be built and run")
```

Comment:
nit: message could tell user how to fix by saying "don't set OCOS_ENABLE_CTEST when using OCOS_ENABLE_SELECTED_OPLIST" instead of just saying it's broken.

or alternatively we could print a warning and unset OCOS_ENABLE_CTEST

Reply:
OCOS_ENABLE_CTEST is a cmake option and we should not update the value.
Will refine the message in another PR for that.